spec/index.adoc

```diff
@@ -1560,6 +1567,12 @@ Calls to a frozen canister will be rejected (like for a stopping canister). Addi
 +
 Default value: 2592000 (approximately 30 days).
 
+* `properties` (`vec record { name: text; value: opt blob; }`)
++
+A set of arbitrary named properties with optional blob values.  The total size of the properties must be less than 4KB.
```

Comment:
```suggestion
* `properties` (`vec record { name: text; value: blob; }`)
+
A set of arbitrary named properties with arbitrary values. The names in the list must be free of duplicates. See `_state-tree-canister-information`.
```
(I forgot the precise syntax for linking within asciidoc, sorry.)

Since the conceptual data model here needs to map to state trees, and state trees don’t have optional values, we should not use them here either.



Is it a problem that you can only set all of them at once? With canister settings, we can set one value without having to know the others.